samila/functions.py

```diff
@@ -217,6 +218,7 @@ def generate_params_filter(
         seed = g.seed
         if g.seed is None:
             seed = random.randint(SEED_LOWER_BOUND, SEED_UPPER_BOUND)
+    g.seed, g.start, g.step, g.stop = seed, start, step, stop
     return seed, start, step, stop
```

Comment:
Do we need this line?

Reply:
If you mean the `g.seed, g.start, g.step, g.stop = seed, start, step, stop`, yes we need this because if we do not update the attributes here when we are saving config or data the old values of attributes will be saved.

And about the next line, we return these values to use them in caller function.
You mean we don't need to return anything so we'll work with attributes, right?